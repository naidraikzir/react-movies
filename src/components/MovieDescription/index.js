import React from 'react';
import {
  Box,
  Flex,
  Tag,
  TagLabel,
  Text,
  chakra,
  useColorModeValue,
} from '@chakra-ui/react';

const MovieDescription = ({
  Title,
  Year,
  imdbRating,
  Genre,
  Plot,
  Actors,
}) => {
  const detailBg = useColorModeValue('rgba(255, 255, 255, 0.5)', 'gray.800');

  return (
    <Box
      bg={detailBg}
      p={{ sm: 8 }}
      py={{ base: 4, sm: 8 }}
      rounded={{ md: '2xl' }}
      shadow={{ md: 'xl' }}
      pos="relative"
      left={{ md: '-5em' }}
      top={{ base: '-20em', md: '5em' }}
      sx={{ backdropFilter: 'blur(20px)' }}
    >
      <Text
        fontSize={['2xl', '3xl', '4xl']}
        fontWeight="bold"
      >
        {Title} {(
          <chakra.span
            fontSize="0.8em"
            opacity="0.8"
            ml="2"
          >
            {Year}
          </chakra.span>
        )}
      </Text>

      <Flex
        alignItems="center"
        mt="4"
      >
        <chakra.span
          fontSize="xl"
          mr="6"
        >
          ⭐️ {imdbRating} / <chakra.sub>10</chakra.sub>
        </chakra.span>
        <Tag
          size="sm"
          variant="outline"
          colorScheme="orange"
        >
          <TagLabel>{Genre}</TagLabel>
        </Tag>
      </Flex>

      <Text mt="6">
        {Plot}
      </Text>

      <Box mt="14">
        <Text>Stars: {Actors}</Text>
      </Box>
    </Box>
  );
};

export default MovieDescription;
