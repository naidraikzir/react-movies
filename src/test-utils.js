import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ChakraProvider, theme } from '@chakra-ui/react';
import reducer from 'store/reducer';

const customRender = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...options
  } = {}
) => {
  const AllProviders = ({ children }) => (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
  return render(ui, { wrapper: AllProviders, ...options });
}

export { customRender as render };
