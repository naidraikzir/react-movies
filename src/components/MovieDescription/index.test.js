import React from 'react';
import { render } from 'test-utils';
import MovieDescription from '.';

const props = {
  Title: 'Title',
  Year: 2021,
  imdbRating: '10',
  Genre: 'Genre',
  Plot: 'Lorem ipsum',
  Actors: 'John Wick',
};

test('match snapshot', () => {
  const { container } = render(<MovieDescription {...props} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders description properties', () => {
  const { getByText } = render(<MovieDescription {...props} />);
  expect(getByText(props.Title)).toBeInTheDocument();
  expect(getByText(props.Year)).toBeInTheDocument();
  expect(getByText(props.imdbRating)).toBeInTheDocument();
  expect(getByText(props.Genre)).toBeInTheDocument();
  expect(getByText(props.Plot)).toBeInTheDocument();
  expect(getByText(`Stars: ${props.Actors}`)).toBeInTheDocument();
});
