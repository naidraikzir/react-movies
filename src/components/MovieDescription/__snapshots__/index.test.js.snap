// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  class="css-1urwz0x"
>
  <p
    class="chakra-text css-1ewjzqi"
  >
    Title
     
    <span
      class="css-65vpd2"
    >
      2021
    </span>
  </p>
  <div
    class="css-fd0k57"
  >
    <span
      class="css-1al1bi6"
    >
      ⭐️ 
      10
       / 
      <sub
        class="css-0"
      >
        10
      </sub>
    </span>
    <span
      class="css-m800dj"
    >
      <span
        class="css-1h8gyx6"
      >
        Genre
      </span>
    </span>
  </div>
  <p
    class="chakra-text css-1w2p18d"
  >
    Lorem ipsum
  </p>
  <div
    class="css-1adflju"
  >
    <p
      class="chakra-text css-0"
    >
      Stars: 
      John Wick
    </p>
  </div>
</div>
`;
